### ISSUE TITLE
Passing wrong strings for scheduler interval doesn't throw an error

### ISSUE THREAD

**Issue by finnBsch** [PyTorchLightning/pytorch-lightning#5548]
I noticed that when you pass an interval as scheduler parameter that is not 'step' or 'epoch' (for example 'batch') there's no error, it just trains without ever calling the scheduler. Had to look into the source code because I didn't find much info about it in the documentation

**Comment by Borda** [comment]
@finnBsch that is a good observation, mind sending a PR with adding assert or check/raise with exception?

**Comment by finnBsch** [comment]
I am not sure where the argument gets assigned.. I just know that there is no check for it in `update_learning_rates`, however I am sure that it gets assigned somewhere earlier but I cannot find it.

**Comment by rohitgr7** [comment]
@finnBsch here is the code you need to update: https://github.com/PyTorchLightning/pytorch-lightning/blob/8cb2451303fac5d57c79115697e6426f307e8c2c/pytorch_lightning/trainer/optimizers.py#L97
